codegenerator/cli/templates/static/codegen/src/LoadLayer.res

```diff
@@ -31,20 +29,24 @@ module LoadActionMap = {
   }
 
   type key = string
-  type t<'entity> = {
-    singleEntities: dict<array<loadSingle<'entity>>>,
-    lookupByIndex: dict<loadIndex<'entity>>,
+  type t = {
+    singleEntities: dict<array<loadSingle<Entities.internalEntity>>>,
+    lookupByIndex: dict<loadIndex<Entities.internalEntity>>,
+    entityMod: module(Entities.InternalEntity),
+    logger: Pino.t,
```

Comment:
The problem with hoisting the logger up is that logger actually contains a context for the logging. So in other words if we hand down the logger from a given handler/loader it will contain info about which event was being handled on the actual log. So it's probably better to put it on the action

Reply:
Hmm, I'll think about it

Reply:
> Hmm, I'll think about it

It's pretty important for debugging that when you get a log/error it prints out info of which event it came from on which chain